sdk/cosmos/src/operations/create_user.rs

```diff
@@ -0,0 +1,64 @@
+use crate::headers::from_headers::*;
+use crate::prelude::*;
+use azure_core::{
+    collect_pinned_stream,
+    headers::{etag_from_headers, session_token_from_headers},
+    Request as HttpRequest, Response as HttpResponse,
+};
+
+#[derive(Debug, Clone, Default)]
+pub struct CreateUserOptions {
+    consistency_level: Option<ConsistencyLevel>,
+}
+
+impl CreateUserOptions {
+    pub fn new() -> Self {
+        Self {
+            consistency_level: None,
+        }
+    }
+
+    setters! {
+        consistency_level: ConsistencyLevel => Some(consistency_level),
+    }
+
+    pub(crate) fn decorate_request(
+        &self,
+        request: &mut HttpRequest,
+        user_name: &str,
+    ) -> Result<(), crate::Error> {
+        azure_core::headers::add_optional_header2(&self.consistency_level, request)?;
+        let body = CreateUserBody { id: user_name };
+        request.set_body(bytes::Bytes::from(serde_json::to_string(&body)?).into());
+        Ok(())
+    }
+}
+
+#[derive(Serialize, Debug)]
+struct CreateUserBody<'a> {
+    id: &'a str,
+}
+
+#[derive(Debug, Clone)]
+pub struct CreateUserResponse {
```

Comment:
Previously Create, Get, and Replace all shared CreateUserResponse. I separated this out into `CreateUserResponse`, `GetUserResponse`, and `ReplaceUserResponse`. Is this ok, or should there be a shared `UserResponse`?

Reply:
I like the idea of sharing the response, I suppose the only downside is if the response changes in the future for one but not the others. Follow-up question, where should I put the shared `UserResponse`? There is already `cosmos/src/resources/user.rs` or I can create a new `user` module under 'resources'.

Reply:
For Azure SDKs we generally share requests i.e. input but not responses i.e. output. At least in OO or duck-type languages you can always pass something more derived in but getting something more derived out can be problematic (how does the caller know they can upcast safely)?

When in doubt, I'd take a look at our Azure/azure-sdk-for-net (or -java or -cpp) repos for consistency with those. Much discussion was already done about those trying to make it work for all languages as much as possible.

Reply:
It looks like in Azure/azure-sdk-for-java they use a shared [`CosmosUserResponse`](https://github.com/Azure/azure-sdk-for-java/blob/main/sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/models/CosmosUserResponse.java). I couldn't find anything in Azure/azure-sdk-for-net but I did find that in Azure/azure-cosmos-dotnet-v3 they use a shared [`UserResponse`](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/master/Microsoft.Azure.Cosmos/src/Resource/User/UserResponse.cs).

Reply:
Any repos besides Azure/azure-sdk-for-* are old and should not be used as examples. The Azure SDK team was created to unify the different languages (idiomatically) for a consistent API, among other things like mitigating breaking changes from purely REST-generated source.

In .NET, we typically return a `Response<T>`, so if you find the method you might find something like `Response<CosmosUser>` (shipping a class called simply "User" would never pass our archboard, per guidelines). Other languages do something similar.

Reply:
@tg-msft do you know what is ideal here? I haven't really been following all the Cosmos discussions and not even sure where to find source. Is it still track 1?

Reply:
https://apiview.dev/Assemblies/Review/0de57f00b1ad4fdab2fa218f59863c78 is the shape of the API that Krzysztof has been working with them on, but it's not GA'ed yet.  I think the source lives at https://github.com/Azure/azure-cosmos-dotnet-v3/tree/v4.